### ISSUE TITLE
Bogus COMMAND (process name) in --proc/--proc-all output

### ISSUE THREAD

**Issue by Maryse47** [slimm609/checksec.sh#111]
## Issue
`checksec --proc/--proc-all` always shows `<PID>ame`, like `474ame` as `COMMAND` instead of actual process name (it doesn't change if checksec is run as root)

## Debug Report
```
***** Checksec debug *****
uid=1000(user) gid=100(users) groups=100(users)
Linux localhost 5.2.2-1 #1 SMP x86_64 GNU/Linux
checksec version: 2.0.1 -- 2019072401
OS=Arch Linux
VER=
-rwxr-xr-x 1 root root 38952 Mar 11 07:29 /usr/bin/cat
/usr/bin/cat: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=c64391c98718402ae35e8e8c7ece01d1008d65c3, stripped
lrwxrwxrwx 1 root root 4 Jan 10  2019 /usr/bin/awk -> gawk
-rwxr-xr-x 2 root root 674416 Jan 10  2019 /usr/bin/gawk
/usr/bin/gawk: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=2df73d8bce3c99c582df1be3ec5d80eecb3e148f, stripped
-rwxr-xr-x 1 root root 22360 Jun  1  2018 /usr/bin/sysctl
/usr/bin/sysctl: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=2a9b00728f1458714f686bb4dd52e57f832d9780, stripped
-rwxr-xr-x 1 root root 38920 Mar 11 07:29 /usr/bin/uname
/usr/bin/uname: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=27513d78d58e9c8c2c10853146c87d3613611935, stripped
-rwxr-xr-x 1 root root 42984 Mar 11 07:29 /usr/bin/mktemp
/usr/bin/mktemp: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=3360bac8992c0829f5ce045d760a1406fc48f46f, stripped
-rwxr-xr-x 1 root root 724456 May 28 18:23 /usr/bin/openssl
/usr/bin/openssl: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=c196ebcf032832cc6e1f8f73c80ddfce7be8f6c3, for GNU/Linux 3.2.0, stripped
-rwxr-xr-x 1 root root 157808 Dec 21  2018 /usr/bin/grep
/usr/bin/grep: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=efbba1ea6419232c285a239a0bc4cf3608ad7e91, stripped
-rwxr-xr-x 1 root root 83976 Mar 11 07:29 /usr/bin/stat
/usr/bin/stat: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=41da3dc04601869c5a10e65e7a09dd62628f7141, stripped
-rwxr-xr-x 1 root root 26720 May 28 22:21 /usr/bin/file
/usr/bin/file: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=e22e4fa56cce9232226b072638005740afe7f709, for GNU/Linux 3.2.0, stripped
-rwxr-xr-x 1 root root 216400 Nov  3  2018 /usr/bin/find
/usr/bin/find: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=8b1b8e2dd522bd895bd1f184c944e6050d7d2d42, stripped
-rwxr-xr-x 1 root root 47112 Mar 11 07:29 /usr/bin/head
/usr/bin/head: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=635a74b82bea7d6dbe5da08546cd762ff99c0028, stripped
-rwxr-xr-x 1 root root 129088 Jun  1  2018 /usr/bin/ps
/usr/bin/ps: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=864dd7723b4bd8a0cc9d5cac70694312691b4970, stripped
-rwxr-xr-x 1 root root 47080 Mar 11 07:29 /usr/bin/readlink
/usr/bin/readlink: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=08b43723dcb8744d4cc390f55fbdaed8964677e0, stripped
-rwxr-xr-x 1 root root 38888 Mar 11 07:29 /usr/bin/basename
/usr/bin/basename: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=6e16307f993e641fe23605a6555cca17a8f104a9, stripped
-rwxr-xr-x 1 root root 43016 Mar 11 07:29 /usr/bin/id
/usr/bin/id: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=7735e500072c4ecef5161f85f3368e668e9a7679, stripped
-rwxr-xr-x 1 root root 31288 Nov  3  2018 /usr/bin/which
/usr/bin/which: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=02c822d2f01f85aa78a5ff80c65cb3b505d1c6b6, stripped
-rwxr-xr-x 1 root root 182096 Jul 19 13:53 /usr/bin/curl
/usr/bin/curl: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=14b81cba46c65ce0bdeb59dda10ef278d826bc2a, for GNU/Linux 3.2.0, stripped
-rwxr-xr-x 1 root root 616752 May 22 15:37 /usr/bin/readelf
/usr/bin/readelf: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=a29fcb63ca298977ca98a39f774b63ff630f2418, for GNU/Linux 3.2.0, stripped
-rwxr-xr-x 1 root root 255968 Feb 26 10:26 /usr/bin/eu-readelf
/usr/bin/eu-readelf: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, for GNU/Linux 3.2.0, BuildID[sha1]=ffbed77fd2756871aa5d139924e9817d74a66092, stripped
```

## Command run to produce the error
`checksec --proc=systemd`

## OS version and Kernel version
Arch Linux
Linux 5.2.2

## Debug output

```
checksec --debug --proc=systemd
***function isString
* System-wide ASLR***function aslrcheck

***function aslrcheck->PAX ASLR

***function aslrcheck->randomize_va_space
 (kernel.randomize_va_space): Full (Setting: 2)

  Description - Make the addresses of mmap base, heap, stack and VDSO page randomized.
  This, among other things, implies that shared libraries will be loaded to random 
  addresses. Also for PIE-linked binaries, the location of code start is randomized.

  See the kernel file 'Documentation/sysctl/kernel.txt' for more details.

* Does the CPU support NX: 
***function nxcheck
Yes

         COMMAND    PID RELRO           STACK CANARY            SECCOMP          NX/PaX        PIE                     FORTIFY
            1ame      1 ***function root_privs
No read permissions for '/proc/1/exe' (run as root).
```

**Comment by slimm609** [comment]
i see the issue and am adding some other stuff currently and will get this fix in with this next push.

**Comment by slimm609** [comment]
This has been resolved in the code, will close upon merge.   

Thanks for reporting

**Comment by slimm609** [comment]
merged in 2.1.0
